H/HDF5/build_tarballs.jl

```diff
@@ -1,90 +1,598 @@
+    "$(../saved/get_config_setting PAC_FORTRAN_NATIVE_DOUBLE_KIND ../saved/config.status)" \
+    "$(../saved/get_config_setting PAC_FORTRAN_NATIVE_DOUBLE_SIZEOF ../saved/config.status)" \
+    "$(../saved/get_config_setting HAVE_Fortran_INTEGER_SIZEOF ../saved/config.status)" \
+    "$(../saved/get_config_setting FORTRAN_HAVE_C_LONG_DOUBLE ../saved/config.status)" \
+    "$(../saved/get_config_setting FORTRAN_C_LONG_DOUBLE_IS_UNIQUE ../saved/config.status)" \
+    "$(../saved/get_config_setting H5CONFIG_F_NUM_RKIND ../saved/config.status)" \
+    "$(../saved/get_config_setting H5CONFIG_F_RKIND ../saved/config.status)" \
+    "$(../saved/get_config_setting H5CONFIG_F_RKIND_SIZEOF ../saved/config.status)" \
+    "$(../saved/get_config_setting H5CONFIG_F_NUM_IKIND ../saved/config.status)" \
+    "$(../saved/get_config_setting H5CONFIG_F_IKIND ../saved/config.status)"
+
+# Patch the generated `Makefile`:
+# (We could instead patch `Makefile.in`, or maybe even `Makefile.am`.)
+# - HDF5 would also try to build and run `H5detect` to collect ABI information.
+#   We know this information, and thus can provide it manually.
+# - HDF5 would try to build and run `H5make_libsettings` to collect
+#   build-time information. That information seems entirely optional, so
+#   we do mostly nothing instead.
+atomic_patch -p1 ${WORKSPACE}/srcdir/patches/src-Makefile.patch
+atomic_patch -p1 ${WORKSPACE}/srcdir/patches/fortran-src-Makefile.patch
+atomic_patch -p1 ${WORKSPACE}/srcdir/patches/hl-fortran-src-Makefile.patch
+
+# `AM_V_P` is not defined. This must be a shell command that returns
+# true or false depending on whether `make` should be verbose. This is
+# probably caused by a bug in automake, or in how automake was used.
+make -j${nproc} AM_V_P=: "${FLAGS[@]}"
+
+make install
+
+popd
+
+fi
+
+install_license COPYING
+"""
+
+augment_platform_block = """
+    using Base.BinaryPlatforms
+    $(MPI.augment)
+    augment_platform!(platform::Platform) = augment_mpi!(platform)
 """
 
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
-platforms = [
-    Platform("x86_64", "linux"),
-    Platform("aarch64", "linux"; libc="glibc"),
-    Platform("x86_64", "macos"),
-    Platform("x86_64", "windows"),
-    Platform("i686", "windows"),
-    Platform("aarch64", "macos"),
-]
+platforms = supported_platforms()
+platforms = expand_cxxstring_abis(platforms)
+platforms = expand_gfortran_versions(platforms)
+
+platforms, platform_dependencies = MPI.augment_platforms(platforms; MPItrampoline_compat="5.3.0")
+
+# Avoid platforms where the MPI implementation isn't supported
+# OpenMPI
+platforms = filter(p -> !(p["mpi"] == "openmpi" && arch(p) == "armv6l" && libc(p) == "glibc"), platforms)
+# MPItrampoline
+platforms = filter(p -> !(p["mpi"] == "mpitrampoline" && libc(p) == "musl"), platforms)
+platforms = filter(p -> !(p["mpi"] == "mpitrampoline" && Sys.isfreebsd(p)), platforms)
+
+# Platforms:
```

Comment:
Can we remove this long list? It doesn't look very useful, and it's _very_ long

Reply:
Done.